cf-oscore/src/main/java/org/eclipse/californium/oscore/ContextRederivation.java

```diff
@@ -575,9 +575,9 @@ private static void printStateLogging(OSCoreCtx ctx) {
 
 		String output = "Context re-derivation phase: " + currentPhase + " (" + supplemental + ")";
```

Comment:
If the output is "assembled" with in the "debug" or "trace", the string is not created in vain. If you consider that as "code duplicate", you may also use a `LOGGER.isInfoEnabled()` to guard this lines.

Reply:
Hmm, what about something like in commit 4a06fd47ed6343e508b3b3b06b31de08e20c10e9 ?

Reply:
The approach in 4a06fd4 works for me

Reply:
> The approach in [4a06fd4](https://github.com/eclipse/californium/commit/4a06fd47ed6343e508b3b3b06b31de08e20c10e9) works for me

Great! Then I resolve this convo